### PR TITLE
Selective route permission to use embeds, fixes #322 in a better way

### PR DESCRIPTION
**Pull request by rigelk**
note: this time it should work :sweat_smile:

### CODE REVIEW

File: support/nginx/peertube [hunk 1]
```diff
@@ -38,6 +38,8 @@ server {
   # resolver_timeout 5s;
 
   add_header Strict-Transport-Security "max-age=63072000; includeSubDomains; preload";
+  add_header X-Frame-Options DENY;
+  add_header X-Frame-Options https://peertube.example.com/videos/embed;
```
Comment (rezonant) [inline support/nginx/peertube]: If people forget to update this, the default position will be to deny embeds. I think the default should be to allow embeds. This header should be applied dynamically in the codebase not in the server configuration.

Comment (rigelk) [reply support/nginx/peertube]: Should be corrected.

File: server.ts [hunk 2]
```diff
@@ -124,7 +132,10 @@ const apiRoute = '/api/' + API_VERSION
 app.use(apiRoute, apiRouter)
 
 // Services (oembed...)
-app.use('/services', servicesRouter)
+app.use('/services', servicesRouter, helmet.frameguard({
+  action: 'allow-from',
+  domain: CONFIG.WEBSERVER.URL
+}))
```
Comment (rezonant) [inline server.ts]: The regular HTML embed is served from /embed, this is the oembed part

**Comment by rigelk** [comment]
@rezonant I've added a CSP since helmet makes adding one quite easy. Could you have a look at it?

**Pull request merged**